mock/mock.go

```diff
@@ -252,18 +252,16 @@ func (m *Mock) findClosestCall(method string, arguments ...interface{}) (bool, *
 	return true, closestCall
 }
 
-func callString(method string, arguments Arguments, includeArgumentValues bool) string {
-
-	var argValsString string
-	if includeArgumentValues {
-		var argVals []string
-		for argIndex, arg := range arguments {
-			argVals = append(argVals, fmt.Sprintf("%d: %#v", argIndex, arg))
+func callString(method string, arguments Arguments) string {
+	var argVals []string
+	for i, arg := range arguments {
+		if i > 0 {
+			argVals = append(argVals, ",")
 		}
-		argValsString = fmt.Sprintf("\n\t\t%s", strings.Join(argVals, "\n\t\t"))
+		argVals = append(argVals, fmt.Sprintf("%#v", arg))
```

Comment:
For me that argument doesn't make a lot of sense. I could argue most numbers used by developers are smaller than 1000, but that doesn't mean making something that can only handle numbers up to 999 is a good idea, even if it makes things "simpler" for "most" people.

I can come up with plenty of functions returning a []byte that do not contain plain text like:
- [`x509.CreateCertificate`](CreateCertificate)
- [`elf.section.Data`](https://golang.org/pkg/debug/elf/#Section.Data)
- Good ol' `io.Reader.read`, arguably the most versatile interace in golang, used for everything from [bzip2](https://golang.org/pkg/compress/bzip2/#NewReader) to [png](https://golang.org/pkg/image/png/#Decode), and yes, also for text.

Also, when I'd pass `[]byte{0x62, 0x61, 0x72}` to a function, and some error tells me I passed a string containing `bar`, I don't consider that expected or something I'd want. Let alone an error spewing out a bunch of random crapp when I passed binary data to a function expecting, and using []byte for this is perfectly valid for this, binary data.

As a last argument: the `callString` function show arguments to mocked functions, not return types. Any sane function doing something with text should take a `string` (or even better, a [`text.Scanner`](https://golang.org/pkg/text/scanner/)), not a []byte. So while `json.Marshal` might return a []byte, if you want to make a function that does something with the marshalled text, it should take a string, and the output of `json.Marshal` should be wrapped with a string().

Trust me, as someone having to deal with `char *` and `uint8_t *` on a daily basis, be glad Go has a special type for text.